packages/langium/src/parser/langium-parser.ts

```diff
@@ -231,18 +233,39 @@ export class LangiumParser extends AbstractLangiumParser {
         }
     }
 
+    private processHiddenTokens(lexerResult: LexerResult): Array<IToken> { // O(n) where n is the number of tokens
+        const { tokens, hidden } = lexerResult;
+        this.commentProvider.clearComments();
+        let j = 0;
+        for (let i = 0, last = 0, lastId = 0; i < tokens.length; ) {
+            if (j == hidden.length || tokens[i].startOffset < hidden[j].startOffset) {
+                const token = tokens[i++];
+                let payload = token.payload = hidden.slice(last, j);
+                last = j; // save index for immediate comments
+                if (['entry', 'ID'].includes(token.tokenType.name)) {
```

Comment:
Langium can never make any assumptions about the names used by the adopter language.